Scripts/validate-adapter-and-partner-versions.rb

```diff
@@ -0,0 +1,15 @@
+# Validates that an adapter and partner version strings match.
+
+# Parse the version strings from the arguments
+abort "Missing argument. Requires: adapter version string, partner version string." unless ARGV.count == 2
+adapter_version = ARGV[0]
+partner_version = ARGV[1]
+
+# Strip the Chartboost Mediation SDK digit and the adapter build digits
+partner_digits_in_adapter_version = adapter_version.split('.')[1...-1].join('.')
+
+# Strip the cocoapods optimistic operation from the partner version if it exists
+partner_version = partner_version.delete_prefix('~> ')
```

Comment:
Since we require entry of the optimistic operator (which I think is good), can we validate in one of these steps that the entire version string is valid? I'm thinking if someone enters `~>         4.1.0.0.0` or `~>4.1.0.0.0`, those should be considered invalid, since it should have exactly one space between the operator and version (not sure if CocoaPods is that picky, but we should keep it consistent nonetheless). Trailing and leading whitespace can be trimmed automatically though.

Alternatively, we could split the two, trim whitespace, then rejoin them when actually writing to the spec.

Reply:
Added another step that validates the version is correct. I just fail if the format is not exactly what we require to avoid having to sanitize the value everywhere it is used.